test/conftest.py

```diff
@@ -143,15 +144,15 @@ def yaml_loader():
 
 
 @pytest.fixture(scope="module")
-def generate_test_segments():
+def generate_test_segments()->Callable[[List[str]], List[RawSegment]]:
     """Roughly generate test segments.
 
     This is a factory function so that it works as a fixture,
     but when actually used, this will return the inner function
     which is what you actually need.
     """
 
-    def generate_test_segments_func(elems):
+    def generate_test_segments_func(elems:List[str])->List[RawSegment]:
```

Comment:
i ended up not using this fixture but i did want to ask, what is the point of this wrapper and using it as a fixture, rather than just calling the inner function? there's no state to deal with and nothing in the closure so it seems unnecessary

Reply:
I see no reason for it, either. Usually, you would do this if the outer function takes parameters and the inner function is a closure, using those parameters. Perhaps it was done for consistency with other fixtures? Or from not being aware of the "why" and copying from an example?

Reply:
ok, i will leave it as is in this PR but would you be open to a separate PR to eliminate the wrapper if i get ambitious?

Reply:
Sure!

Reply:
I'm fine with you removing this fixture in a separate PR as well. For this PR can you revert this file to keep diffs focused please.